index.js

```diff
@@ -31,7 +31,7 @@ module.exports = function(url) {
         service(urlPieces).then((urls) => {
           // Covnvert to an array if we weren't passed one
           resolve(Array.isArray(urls) ? urls : [ urls ]);
-        });
+        }).catch( err => reject( 'service error: '+err))
```

Comment:
I think it's safe to just pass `reject` as the handler to `catch` by itself. Consumers can do with the error whatever they need. Also, semi-colon at the end please :)

Reply:
hi dxprog. I've added the semicolon.

Dunno about the other bit. You mean change each service handler to use the initial `reject`? I'm not smart or motivated enough to do that :)

Reply:
Ah, the line would just become `.catch(reject);`. The only real difference is you'd get `err` without the `service error` part concatenated, which will be good if `err` is something other than a string. It doesn't really change the way you interact with it, but gives more flexibility.

Reply:
Okay, updated. Thanks for explaining. It makes sense now that I've thought about it.